tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,131 @@
+import tl = require('azure-pipelines-task-lib/task');
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+import path = require('path');
+import armStorage = require('azure-pipelines-tasks-azure-arm-rest/azure-arm-storage');
+import { AzureRMEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azure-arm-endpoint';
+import { AzureEndpoint, StorageAccount } from 'azure-pipelines-tasks-azure-arm-rest/azureModels';
+import * as telemetry from "azure-pipelines-tasks-utility-common/telemetry";
+import * as os from 'os';
+import * as url from 'url';
+import * as fs from 'fs';
+import helpers = require('common-helpers/helpers');
+
+const HELPER_SCRIPT = path.join(__dirname, 'PublishAVD.ps1');
+const HELPER_SCRIPT_EXCEPTIONS_TELEMETRY = path.join(__dirname, 'ReportExceptionTelemetry.ps1');
+const TARGET_DLL = path.join(__dirname,'node_modules/common-helpers/lib/AppAttachFrameworkDLL/AppAttachKernel.dll');
+const TARGET_DLL_TELEMETRY = path.join(__dirname, 'node_modules/common-helpers/lib/AppAttachFrameworkDLL/AppAttachTelemetry.dll');
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+	if (isNonEmpty(resourceUri)) {
+		const parsedUrl = url.parse(resourceUri, true);
+
+		const pathname = parsedUrl.pathname || '';
+		const segments = pathname.split('/');
+
+		const resourceGroupIndex = segments.indexOf('resourceGroups');
+
+		if (resourceGroupIndex !== -1 && resourceGroupIndex < segments.length - 1) {
+			return segments[resourceGroupIndex + 1];
+		}
+	}
+	
+    return "";
```

Comment:
Does everything else work fine when we return empty here?

Reply:
Fetching storage account keys would fail and subsequently caught in exception and task would fail which is expected

Reply:
If it will fail eventually, would it make sense to just throw an exception here and fail immediately? That way we don't risk the bad data having unexpected effects when we use it later on

Reply:
Done